### PR TITLE
Update CHANGELOG and Deprecations with v1.5 changes

### PR DESCRIPTION
**Pull request by eksperimental**
This makes both documents upto date

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -61,18 +61,19 @@
...
   * [Kernel] `:as_char_lists value` in `Inspect.Opts.t/0` type, in favor of `:as_charlists`
   * [Kernel] `:char_lists` key in `Inspect.Opts.t/0` type, in favor of `:charlists`
   * [Module] Using Erlang parse transforms via `@compile {:parse_transform, _}` is deprecated
   * [String] `String.ljust/3` and `String.rjust/3` are deprecated in favor of `String.pad_leading/3` and `String.pad_trailing/3` with a binary padding
   * [String] `String.strip/1` and `String.strip/2` are deprecated in favor of `String.trim/1` and `String.trim/2`
   * [String] `String.lstrip/1` and `String.rstrip/1` are deprecated in favor of `String.trim_leading/1` and `String.trim_trailing/1`
   * [String] `String.lstrip/2` and `String.rstrip/2` are deprecated in favor of `String.trim_leading/2` and `String.trim_trailing/2` with a binary as second argument
-  * [Typespec] `char_list/0` type is deprecated in favor of `charlist/0`
+  *[Typespec] `char_list/0` type is deprecated in favor of `charlist/0`
```
Comment (josevalim) [inline CHANGELOG.md]: Please add the space back? :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -61,18 +61,19 @@
 
 #### Elixir
 
+  * `Atom.to_char_list/1`, `Float.to_char_list/1`, `Integer.to_char_list/1`, `Integer.to_char_list/2`, `Kernel.to_char_list/1`, `List.Chars.to_char_list/1`, `String.to_char_list/1` have been deprecated in favor of their `to_charlist` version.
+    This aligns with the naming conventions in both Erlang and Elixir
```
Comment (josevalim) [inline CHANGELOG.md]: Please keep everything in a single line.

File: CHANGELOG.md (hunk 1 above)
Comment (eksperimental) [inline CHANGELOG.md]: oops

**Comment by eksperimental** [comment]
@josevalim ready to merge

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**